packages/schematics/src/add/index.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Rule, SchematicContext, SchematicsException, Tree, chain } from "@angular-devkit/schematics";
+import { Path, normalize } from '@angular-devkit/core';
+import { NodePackageInstallTask } from "@angular-devkit/schematics/tasks";
+import { addSymbolToNgModuleMetadata } from "@schematics/angular/utility/ast-utils";
+import { InsertChange } from "@schematics/angular/utility/change";
+import { Schema as ComponentOptions } from "./schema";
+
+import * as ts from "typescript";
+import * as fs from "fs";
+
+// Looks up and finds the path to the app module (or other module if specified)
+function findModuleFromOptions(host: Tree, options: ComponentOptions): Path | undefined {
+    const modulePath = normalize(
+      '/' + options.sourceDir + '/' + (options.appRoot) + '/' + options.module);
+    const moduleBaseName = normalize(modulePath).split('/').pop();
```

Comment:
Regarding this `.split('/')`, on windows backslashes are used. Would that matter here?

Reply:
As I understand it, we normalize to *nix format and later on its handled for Windows. There is a 'virtual filesystem' in play here, where we apply changes in a file tree and then later merge it with the real file tree (and I believe that is where they handle windows backslash).